llama-index-core/llama_index/core/workflow/workflow.py

```diff
@@ -239,12 +291,9 @@ def get_result(self) -> Any:
     @step()
     async def _done(self, ev: StopEvent) -> None:
         """Tears down the whole workflow and stop execution."""
-        # Stop all the tasks
-        for t in self._tasks:
-            t.cancel()
```

Comment:
Two problems here:
- `_done` would cancel itself (not sure about the actual implications but it's not pretty)
- in case of a workflow timeout we wouldn't pass here, resulting in running tasks at the moment `run` exits